src/test/e2e/tracks.spec.ts

```diff
@@ -0,0 +1,175 @@
+      const url = new URL(route.request().url());
+      const page = parseInt(url.searchParams.get('page') || '1');
+      
+      if (page === 1) {
+        await route.fulfill({
+          status: 200,
+          contentType: 'application/json',
+          body: JSON.stringify({
+            data: [
+              {
+                id: '1',
+                title: 'Track 1',
+                artist: 'Artist 1',
+                album: 'Album 1',
+                genres: ['Rock'],
+                coverImage: 'https://example.com/cover1.jpg',
+                audioFile: 'https://example.com/audio1.mp3',
+              },
+            ],
+            meta: {
+              total: 2,
+              page: 1,
+              limit: 1,
+              totalPages: 2,
+            },
+          }),
+        });
+      } else {
+        await route.fulfill({
+          status: 200,
+          contentType: 'application/json',
+          body: JSON.stringify({
+            data: [
+              {
+                id: '2',
+                title: 'Track 2',
+                artist: 'Artist 2',
+                album: 'Album 2',
+                genres: ['Pop'],
+                coverImage: 'https://example.com/cover2.jpg',
+                audioFile: 'https://example.com/audio2.mp3',
+              },
+            ],
+            meta: {
+              total: 2,
+              page: 2,
+              limit: 1,
+              totalPages: 2,
+            },
+          }),
+        });
+      }
+    });
+    
+    await page.reload();
+    
+    await expect(page.getByTestId('loading-tracks')).not.toBeVisible();
+    
+    await expect(page.getByTestId('pagination')).toBeVisible();
+    
+    await expect(page.getByText('Track 1')).toBeVisible();
+    await expect(page.getByText('Track 2')).not.toBeVisible();
+    
+    await page.getByText('2').click();
+    
+    await expect(page.getByText('Track 2')).toBeVisible();
+    await expect(page.getByText('Track 1')).not.toBeVisible();
+  });
+
+  test('should handle create track button click', async ({ page }) => {
+    await expect(page.getByTestId('loading-tracks')).not.toBeVisible();
+    
+    await page.getByTestId('create-track-button').click();
+    
+  });
+
+  test('should handle responsive design', async ({ page }) => {
+    await page.setViewportSize({ width: 375, height: 667 });
+    
+    await expect(page.getByTestId('loading-tracks')).not.toBeVisible();
+    
+    await expect(page.getByTestId('tracks-header')).toBeVisible();
+    await expect(page.getByTestId('create-track-button')).toBeVisible();
+    await expect(page.getByTestId('search-input')).toBeVisible();
+    
+    await page.setViewportSize({ width: 1920, height: 1080 });
+    
+    await expect(page.getByTestId('tracks-header')).toBeVisible();
+    await expect(page.getByTestId('create-track-button')).toBeVisible();
+    await expect(page.getByTestId('search-input')).toBeVisible();
```

Comment:
It's better to move test ids to a constant. As if the id changes you have to manually fix in all places